src/pfe/portal/modules/LoadRunner.js

```diff
@@ -705,10 +724,36 @@ module.exports = class LoadRunner {
   }
 
   emitCompleted() {
-    if (!this.collectingHCD) {
+    if (!this.status.COLLECTING_HCD) {
       this.user.uiSocket.emit('runloadStatusChanged', { projectID: this.project.projectID,  status: 'completed' });
       clearTimeout(this.heartbeatID);
-      this.project = null;
+      this.resetStatus();
+    }
+  }
+
+  resetStatus() {
+    this.status = {
+      CURRENT_STATE: states.IDLE,
+      REQUESTED: false,
+      PREPARING: false,
+      STARTING: false,
+      STARTED: false,
+      COMPLETED: false,
+      CANCELLED: false,
+    };
+  }
+
+  isIdle() {
+    if (this.status.CURRENT_STATE === states.IDLE) {
+      return true;
+    }
+    return false;
+  }
+
+  setStatus(newStatus) {
+    if (this.status[newStatus] !== null) {
```

Comment:
Can we add a `log.trace()` here to say what we're setting the status to for which project? It would be very useful if we are debugging loadrunner in the future.